### PR TITLE
Implement azredevops_agent_queue resource

### PR DESCRIPTION
**Pull request by nmiodice**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
There is no way to configure an Agent Pool defined at the org, in a specific project.

Issue Number: https://github.com/microsoft/terraform-provider-azuredevops/issues/300

This also fixes
 - an issue where the data source for agent_pool fails if the resource cannot be found, which does not adhere to the Terraform API contract.
 - a code duplication issue caught by sonar cloud

## What is the new behavior?
-------------------------------------
There is now the ability to use org-wide agent pools in a project.

The HCL looks like this:

```hcl
resource "azuredevops_project" "p" {
  project_name = "Sample Project"
}

data "azuredevops_agent_pool" "p" {
  name = "contoso-pool"
}

resource "azuredevops_agent_queue" "q" {
  project_id    = azuredevops_project.p.id
  agent_pool_id = data.azuredevops_agent_pool.p.id
}

# Grant acccess to queue to all pipelines in the project
resource "azuredevops_resource_authorization" "auth" {
  project_id  = azuredevops_project.p.id
  resource_id = azuredevops_agent_queue.q.id
  type        = "queue"
  authorized  = true
}
```

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: azuredevops/resource_resource_authorization.go [hunk 1]
```diff
@@ -33,15 +33,15 @@ func resourceResourceAuthorization() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Description:  "id of the resource",
-				ValidateFunc: validation.IsUUID,
+				ValidateFunc: validation.NoZeroValues,
```
Comment (xuzhang3) [inline azuredevops/resource_resource_authorization.go:36]: Why chang `validation.IsUUID` to `validation.NoZeroValues`

**Comment by xuzhang3** [comment]
I think we can pending this PR until we have `data_agent_queue`.  Currently  users need hard coded `pool_id` .

File: azuredevops/resource_resource_authorization.go (hunk 1 above)
Comment (nmiodice) [reply azuredevops/resource_resource_authorization.go:36]: The authorization when  `type == "endpoint"` assumed that the resource ID was a service endpoint, which uses a `UUID`. Other resources do not. The API does not require one or the other

File: azuredevops/resource_agent_queue.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+				ValidateFunc:     validation.NoZeroValues,
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+		},
+	}
+}
+
+func resourceAgentQueueCreate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*config.AggregatedClient)
+	queue, projectID, err := expandAgentQueue(d)
```
[the remark this answers is not in the document]
Comment (nmiodice) [reply azuredevops/resource_agent_queue.go:49]: I should be checking for the error... oops!

**Comment by nmiodice** [comment]
@xuzhang3 I think that this can still be used if the referenced pool was configured using `auto_provision = false` or if the ID is found using an Azure CLI call.

https://github.com/microsoft/terraform-provider-azuredevops/blob/master/website/docs/r/agent_pool.html.markdown

**Comment by xuzhang3** [comment]
> @xuzhang3 I think that this can still be used if the referenced pool was configured using `auto_provision = false` or if the ID is found using an Azure CLI call.
> 
> https://github.com/microsoft/terraform-provider-azuredevops/blob/master/website/docs/r/agent_pool.html.markdown

I still think we should hold this PR until data_soruce is ready. @EliiseS Have any update for #293?

**Comment by EliiseS** [comment]
It's almost done, I'll try to get a PR out tonight or tomorrow my time.

**Comment by EliiseS** [comment]
Here's the PR for agent pool data source: #339

**Comment by nmiodice** [comment]
cc @xuzhang3 - this is ready for review

**Pull request merged**